r/R/trajectory_analysis.R

```diff
@@ -100,13 +100,37 @@ runTrajectoryAnalysis <- function(req, data) {
 #'
 #' @return a cell_data_set object with cluster and graph information stored internally
 #' @export
-generateGraphData <- function(data) {
-  cell_data <- SeuratWrappers::as.cell_data_set(data)
-
+generateGraphData <- function(embedding_data, embedding_settings, clustering_settings, data) {
   set.seed(ULTIMATE_SEED)
 
-  cell_data <- monocle3::cluster_cells(cds = cell_data, reduction_method = "UMAP")
-  cell_data <- monocle3::learn_graph(cell_data, use_partition = TRUE)
+  Seurat::DefaultAssay(data) <- "RNA"
+
+  clustering_method <- clustering_settings$method
+  embedding_method <- embedding_settings$method
+
+  # Clustering resolution can only be used by monocle if the clustering method is leiden
```

Comment:
So when the clustering resolution is NULL, it is determined automatically by monocle. What happens if the clustering method is louvain and the resolution chosen by the user is different from the one determined by monocle? Will there be any discrepancies?

Reply:
Looking at the monocle3 documentation and code, the monocle3 will set the resolution automtically. I do not know the reasoning behind this. Setting the resolution value when setting the clustering method to leiden will automatically set the clustering settings to use leiden: https://github.com/cole-trapnell-lab/monocle3/blob/87f6e88760566065179ae5039d524da7d032cc15/R/cluster_cells.R#L118-L122